public/controllers/agent/components/register-agent.js

```diff
@@ -151,6 +208,69 @@ export class RegisterAgent extends Component {
     return text;
   }
 
+  async getGroups() {
+    try {
+      const result = await WzRequest.apiReq(
+        'GET',
+        '/groups',
+        {}
+      );
+      return result.data.data.affected_items.map(item =>
+        ({ label: item.name, id: item.name })
+      )
+    } catch (error) {
+      return [];
+    }
+  }
+
+  optionalDeploymentVariables() {
+    let deployment = `WAZUH_MANAGER='${this.state.serverAddress}' ${this.state.selectedOS == 'win' ? `WAZUH_REGISTRATION_SERVER='${this.state.serverAddress}' ` : ''}${this.state.needsPassword
+      ? ` WAZUH_REGISTRATION_PASSWORD='${this.state.wazuhPassword}'`
+      : ''
+      }${this.state.udpProtocol
+        ? " WAZUH_PROTOCOL='UDP'"
+        : ''
+      }${this.state.selectedGroup.length ? `WAZUH_AGENT_GROUP='${this.state.selectedGroup.map(item => item.label).join(',')}' ` : ''}`
+    return deployment;
+  }
+
+  optionalPackets() {
+    let ret = `https://packages.wazuh.com/4.x/yum5/x86_64/wazuh-agent-${this.state.wazuhVersion}-1.x86_64.rpm`;;
+
+    if (this.state.selectedOS == 'rpm' && this.state.selectedVersion == 'centos5' && this.state.selectedArchitecture == 'i386') {
```

Comment:
to avoid this if-else chain, you can use some specific functions to resolve related data for example
```js
switch(this.state.selectedOS) {
  case 'rpm': 
    return resolveRPMPackage();
  case 'deb': 
    return resolveDEBPackage():
  default: 
   return: `https://packages.wazuh.com/4.x/yum5/x86_64/wazuh-agent-${this.state.wazuhVersion}-1.x86_64.rpm`;
}
```
the extra functions like `resolveRMPPackage()` can be a similar switch approach 
the important is to keep functions easy to read